src/Symfony/Component/Validator/Tests/Mapping/ClassMetadataTest.php

```diff
@@ -308,6 +309,19 @@ public function testGetPropertyMetadataReturnsEmptyArrayWithoutConfiguredMetadat
     {
         $this->assertCount(0, $this->metadata->getPropertyMetadata('foo'), '->getPropertyMetadata() returns an empty collection if no metadata is configured for the given property');
     }
+
+    public function testClassMetadataMembersAreUniqueByPropertyAndByClassAfterMergingConstraints()
+    {
+        $metadata = new ClassMetadata(self::INTERFACE_A_CLASS);
+        $metadata->addGetterConstraint('data', new ConstraintA());
+
+        $otherMetadata = new ClassMetadata(self::INTERFACE_A_CLASS);
+        $otherMetadata->addGetterConstraint('data', new ConstraintB());
+
+        $metadata->mergeConstraints($otherMetadata);
+
+        $this->assertCount(1, $metadata->getPropertyMetadata('data'));
```

Comment:
it's just a test :) the behavior is actually correct imo. Skip constraints from the same class/property (i.e. the ones we already visisted). in real life these are the same ;-)